### PR TITLE
add a plugin for playwright selector engine

### PR DESCRIPTION
**Pull request by Georgegriff**

### CODE REVIEW

**Comment by Georgegriff** [comment]
@ihor-nikiforov-scout I've produced a playwright `plugin` and tested it out, was a little tricky to get the library injected into the function for playwright because they do a .toString on functions to transfer them into the browser scope, but i found a work around

File: plugins/playwright/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const path = require("path");
+
+// load the library in UMD format which self executes and adds window.querySelectorShadowDom
+const querySelectorShadowDomUMD = fs.readFileSync(path.resolve(__dirname, "../../dist/querySelectorShadowDom.js"))
+
+// a string because playwright does a .toString on a selector engine and we need to
+// make sure that query-selector-shadow-dom is injected and loaded into the function closure
+const engineString =`
+    options = options || {};
+    const name = options.name;
```
Comment (ihor-nikiforov-scout) [inline plugins/playwright/index.js]: Please add a default name of the selector engine.
This way we can register the selector engine without having to set an optional name parameter.

File: examples/playwright/custom-engine.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const { selectorEngine } = require("query-selector-shadow-dom/plugins/playwright");
```
Comment (ihor-nikiforov-scout) [inline examples/playwright/custom-engine.js]: This is a beautiful interface

**Comment by ihor-nikiforov-scout** [comment]
@Georgegriff
I've tested these changes in my project and it works like a charm.
Thank you.

**Comment by thernstig** [comment]
Add documentation for how to use this in the README.md?

**Comment by Georgegriff** [comment]
@thernstig updated the readme, and done a new release to 0.4.0 not appeared on npm yet, not sure why

**Comment by thernstig** [comment]
@Georgegriff Just a question. Is there a reason one need to add `shadow=`, couldn't that be the default if one use the plugin? I assume if one uses it, one by default want to pierce all shadow dom?

**Comment by Georgegriff** [comment]
It is the default :)

**Comment by thernstig** [comment]
Oh, it is? But looking at the updated README it says:

```javascript
  // shadow= allows a css query selector that automatically pierces shadow roots.
  await page.waitForSelector('shadow=#no-downloads span', {timeout: 3000})
```

**Comment by Georgegriff** [comment]
Do you mean omit it entirely? Can't do that that's how playwright selector engines work it's how it know where to go, you're free to change `shadow=` to whatever else you want but something= is requires by plauwri5

**Comment by thernstig** [comment]
I should have read the API for `selectors.register` before commenting obviously ;) Would be nice if playwright let you set a default selectorEngine to use. Imagine Custom Element-heavy pages (e.g. the ones using lots of lit-html or maybe Stencil), they most likely want to penentrate the shadow dom in all calls. Worth writing an issue on playwright you think?

**Comment by Georgegriff** [comment]
I've not tried it but registering the plugin with a name of CSS may overwrite their default

**Comment by thernstig** [comment]
@Georgegriff Will test this once I move to playwright (might take a while) and if it does not work I'll make a feature suggestion on the Playwright repo. Keep staying awesome with this plugins, appreciate it! 💯

**Pull request merged**